sdk/tables/azure-data-tables/azure/data/tables/_base_client.py

```diff
@@ -249,7 +248,7 @@ def _configure_policies(self, **kwargs):
     def _configure_credential(self, credential):
         # type: (Any) -> None
         if hasattr(credential, "get_token"):
-            self._credential_policy = BearerTokenCredentialPolicy(  # type: ignore
+            self._credential_policy = BearerTokenChallengePolicy(  # type: ignore
                 credential, STORAGE_OAUTH_SCOPE
```

Comment:
So tables uses the default `True` for the boolean flags?

Reply:
It does -- this works for API versions that support auth challenges as well as those that don't. Tables challenges contain a valid `resource` field in the WWW-Authenticate header, so scopes discovery does work correctly